pkg/services/annotations/annotationsimpl/cleanup_test.go

```diff
@@ -164,6 +162,32 @@ func TestOldAnnotationsAreDeletedFirst(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestAnnotationCleanUp_Timeout(t *testing.T) {
+	annotationCount := 21 // createTestAnnotations works best in multiples of 3
+	fakeSQL := db.InitTestDB(t)
+	cfg := setting.NewCfg()
+	cfg.AnnotationCleanupJobBatchSize = 100 // batch size > total number of annotations
+	createTestAnnotations(t, fakeSQL, annotationCount, 6)
+	cleaner := ProvideCleanupService(fakeSQL, cfg)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Millisecond)
+	defer cancel()
+
+	// configuration to delete all but one of everything
+	cfg.AlertingAnnotationCleanupSetting = settingsFn(0, 1)
+	cfg.DashboardAnnotationCleanupSettings = settingsFn(0, 1)
+	cfg.APIAnnotationCleanupSettings = settingsFn(0, 1)
+
+	go func() {
```

Comment:
Note for reviewers: while this is working just fine and I haven't had a test flap since I added it, I'm not actually positive that it's working the way I think it is. I _believe_ that the `ctx` countdown timer is scoped to the goroutine, so the `time.Sleep` method runs first (thereby guaranteeing that the 1ms timeout has expired).